.eslintrc.json

```diff
@@ -4,15 +4,48 @@
     "es2021": true,
     "node": true
   },
-  "extends": ["eslint:recommended", "plugin:@typescript-eslint/recommended"],
+  "extends": ["eslint:recommended", "plugin:@typescript-eslint/recommended", "plugin:import/typescript"],
   "parser": "@typescript-eslint/parser",
   "parserOptions": {
     "ecmaVersion": "latest",
     "sourceType": "module"
   },
-  "plugins": ["@typescript-eslint", "compat"],
+  "plugins": ["@typescript-eslint", "compat", "import"],
   "rules": {
     "compat/compat": "error",
+    "import/no-restricted-paths": [
+      "error",
+      {
+        "zones": [
+          {
+            "target": "./src/www/infrastructure",
```

Comment:
Could you please add a comment explaining what the rule means?
I have to dig the documentation, and even then they do a terrible job explaining.

I think this means infrasttructure cannot depend on anything in www, except infrastructure.

BTW, TODO: move infrastructure out of www. That's kind of legacy from pre-webpack.

Reply:
should we move model as well? i think src/types should be split up by concern as well.

also, how do you want things set up in the manager? generally the same i assume, but any key differences i should be aware of?

Reply:
I'd leave the model in the app for now. I guess it could make sense to move out if multiple apps were depending on it.

We can discuss the manager when we get to it. I'd rather get the restrictions first. The manger seems a bit messier.
cloud/ should be in infrastructure/. Install_scripts should probably be in the app code, but I think we like to github, so we need to consider breakages. It's still a bit unclear to me how we should think about the Electron and web parts of the app.